packages/@aws-containers/aws-ecs-builder/lib/extensions/appmesh.ts

```diff
@@ -0,0 +1,216 @@
+      // for metadata endpoints or other ports that need direct
+      // communication
+      proxyConfiguration: new ecs.AppMeshProxyConfiguration({
+        containerName: 'envoy',
+        properties: {
+          appPorts: [containerextension.trafficPort],
+          proxyEgressPort: 15001,
+          proxyIngressPort: 15000,
+          ignoredUID: 1337,
+          ignoredGID: 1338,
+          egressIgnoredIPs: [
+            '169.254.170.2', // Allow services to talk directly to ECS metadata endpoints
+            '169.254.169.254', // and EC2 instance endpoint
+          ],
+          // Note that at some point we will need other extensions like
+          // MySQL to be able to add their ports to this egress ignored
+          // ports automatically.
+          egressIgnoredPorts: [
+          ],
+        },
+      }),
+    } as ecs.TaskDefinitionProps;
+  }
+
+  public useTaskDefinition(taskDefinition: ecs.TaskDefinition) {
+    const appMeshRepo = ecr.Repository.fromRepositoryArn(
+      this.scope,
+      `${this.parentService.id}-envoy-repo`,
+      'arn:aws:ecr:us-east-1:840364872350:repository/aws-appmesh-envoy',
+    );
+
+    this.container = taskDefinition.addContainer('envoy', {
+      image: ecs.ContainerImage.fromEcrRepository(appMeshRepo, 'v1.13.1.1-prod'),
+      essential: true,
+      environment: {
+        APPMESH_VIRTUAL_NODE_NAME: `mesh/${this.mesh.meshName}/virtualNode/${this.parentService.id}`,
+        AWS_REGION: cdk.Stack.of(this.parentService).region,
+        ENABLE_ENVOY_STATS_TAGS: '1',
+        ENABLE_ENVOY_DOG_STATSD: '1',
+      },
+      healthCheck: {
+        command: [
+          'CMD-SHELL',
+          'curl -s http://localhost:9901/server_info | grep state | grep -q LIVE',
+        ],
+        startPeriod: cdk.Duration.seconds(10),
+        interval: cdk.Duration.seconds(5),
+        timeout: cdk.Duration.seconds(2),
+      },
+      memoryReservationMiB: 128,
+      user: '1337',
+      logging: new ecs.AwsLogDriver({ streamPrefix: 'envoy' }),
+    });
+
+    // Raise the number of open file descriptors allowed.
+    this.container.addUlimits({
+      softLimit: 1024000,
+      hardLimit: 1024000,
+      name: ecs.UlimitName.NOFILE,
+    });
+  }
+
+  // Enable cloudmap for the service
+  public mutateServiceProps(props: ServiceBuild) {
+    return {
+      ...props,
+
+      // We must make sure that service tasks are registered into
+      // CloudMap so that the App Mesh proxy can find them.
+      cloudMapOptions: {
+        dnsRecordType: 'A',
+        dnsTtl: cdk.Duration.seconds(10),
+        failureThreshold: 2,
+        name: this.parentService.id,
+      },
+
+      // These specific deployment settings are currently required in order to
+      // maintain availability during a rolling deploy of the service with App Mesh
+      // Warning, these settings do not work with a low task count however.
+      // minHealthyPercent: 100,
+      // maxHealthyPercent: 125,
```

Comment:
I think we still want to use custom settings here for App Mesh, see [this document](https://docs.aws.amazon.com/app-mesh/latest/userguide/best-practices.html#reduce-deployment-velocity) for the recommended best practice.

Reply:
Fixed